annif/eval.py

```diff
@@ -145,10 +145,43 @@ def _evaluate_samples(self, y_true, y_pred, metrics='all'):
 
         return results
 
-    def results(self, metrics='all'):
+    def output_result_per_label(self, y_true, y_pred, results_file):
+        """Write results per label (non-aggregated) to outputfile"""
+
+        y_pred = y_pred.T > 0.0
+        y_true = y_true.T > 0.0
+
+        tp = (y_true & y_pred)
+        fp = (~y_true & y_pred)
+        fn = (y_true & ~y_pred)
+
+        r = len(y_true)
+
+        with open(results_file, 'w') as f:
```

Comment:
The result file should probably be opened in UTF-8 mode? Also, please avoid too short variable names. 

If you use a click.File then you don't have to open it here and can just use `results_file` as a file object.